src/tests/Dictionary/DictionaryContainer.test.jsx

```diff
@@ -155,4 +156,79 @@ describe('DictionaryOverview', () => {
     const wrapper = shallow(<MemoryRouter><DictionaryOverview {...props} /></MemoryRouter>);
     expect(wrapper).toMatchSnapshot();
   });
+
+  it('should handle a new version release', () => {
+    const props = {
+      dictionary: [dictionary],
+      versions: [customVersion],
+      dictionaryConcepts: [dictionary],
+      isReleased: false,
+      hideVersionModal: jest.fn(),
+      showVersionModal: jest.fn(),
+      error: null,
+      match: {
+        params: {
+          ownerType: 'testing',
+          owner: 'tester',
+          type: 'collection',
+          name: 'chris',
+        },
+      },
+      fetchDictionary: jest.fn(),
+      fetchVersions: jest.fn(),
+      fetchDictionaryConcepts: jest.fn(),
+      releaseHead: jest.fn(),
+      createVersion: jest.fn(),
+    };
+    jest.useFakeTimers();
+    const wrapper = mount(<Provider store={store}><MemoryRouter><DictionaryOverview {...props} /></MemoryRouter></Provider>);
+    const spy = jest.spyOn(wrapper.find('DictionaryOverview').instance(), 'handleCreateVersion');
+    wrapper.instance().forceUpdate();
+    wrapper.find('.fas.fa-cloud-upload-alt.version').simulate('click');
+    wrapper.find('.btn-sm.btn-outline-info.version').at(0).simulate('click');
+    expect(spy).toHaveBeenCalledTimes(1);
+    jest.runAllTimers();
+  });
+
+  it('should handleOnchange', () => {
+    const props = {
+      dictionary: [dictionary],
+      versions: [customVersion],
+      dictionaryConcepts: [dictionary],
+      isReleased: false,
+      hideVersionModal: jest.fn(),
+      showVersionModal: jest.fn(),
+      error: null,
+      match: {
+        params: {
+          ownerType: 'testing',
+          owner: 'chriskala',
+          type: 'collection',
+          name: 'chris',
+        },
+      },
+      fetchDictionary: jest.fn(),
+      fetchVersions: jest.fn(),
+      fetchDictionaryConcepts: jest.fn(),
+      releaseHead: jest.fn(),
+      createVersion: jest.fn(),
+    };
+    const event = {
+      preventDefault: jest.fn(),
+      target: { 
+        value: 'v2.0', 
+        name: 'versionId' 
+      }
+    };
+
+    jest.useFakeTimers();
+    const wrapper = mount(<Provider store={store}><MemoryRouter><DictionaryOverview {...props} /></MemoryRouter></Provider>);
+    const spy = jest.spyOn(wrapper.find('DictionaryOverview').instance(), 'handleChange');
+    wrapper.instance().forceUpdate();
+    wrapper.find('.fas.fa-cloud-upload-alt.version').simulate('click');
+    expect(wrapper.find('#versionId').at(0).simulate('change', {
```

Comment:
What is this assertion expecting?

Reply:
Thank you for that. This test has been refactored.